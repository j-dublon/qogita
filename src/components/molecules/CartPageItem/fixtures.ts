import { CartPageItemProps } from "@/types";

export const mockHandleUpdateCart = jest.fn();

export const mockProps: CartPageItemProps = {
  item: {
    name: "Mock Product",
    imageUrl: "http://your-product.com",
    recommendedRetailPrice: 52,
    price: 52,
    recommendedRetailPriceCurrency: "GBP",
    gtin: "987",
    id: "987",
    brandName: "Chanel",
    categoryName: "Perfume",
    quantity: 3,
  },
  handleUpdateCart: mockHandleUpdateCart,
};
