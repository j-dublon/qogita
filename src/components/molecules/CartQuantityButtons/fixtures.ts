import { CartQuantityButtonsProps } from "@/types";

export const mockUpdateCartQuantity = jest.fn();

export const mockProps: CartQuantityButtonsProps = {
  product: {
    name: "Mock Product",
    imageUrl: "http://your-product.com",
    recommendedRetailPrice: 52,
    recommendedRetailPriceCurrency: "GBP",
    gtin: "987",
    brandName: "Chanel",
    categoryName: "Perfume",
  },
  numberInCart: 3,
  updateCartQuantity: mockUpdateCartQuantity,
};
