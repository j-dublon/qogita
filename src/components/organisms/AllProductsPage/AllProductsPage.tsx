import React, { FC } from "react";
import Layout from "../../Layout";
import { AllProductsPageProps } from "@/types";
import ResponsivePaginationComponent from "react-responsive-pagination";
import "react-responsive-pagination/themes/classic.css";
import { Loading, Error } from "../../atoms";
import { ProductsList } from "../../molecules";

const AllProductsPage: FC<AllProductsPageProps> = ({
  loading,
  error,
  products,
  currentPage,
  totalPages,
  fetchData,
}) => {
  return (
    <Layout>
      <h1 className="text-3xl font-semibold text-center my-6">Products</h1>
      {loading && <Loading />}
      {error ? (
        <Error />
      ) : (
        <div className="mb-8">
          <ProductsList products={products} />
          <ResponsivePaginationComponent
            current={currentPage}
            total={totalPages}
            onPageChange={(page) => fetchData(page)}
            aria-label="Pagination controls for all products page"
          />
        </div>
      )}
    </Layout>
  );
};

export default AllProductsPage;
