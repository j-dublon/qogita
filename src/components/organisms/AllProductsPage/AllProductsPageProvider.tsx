import React, { FC, useState } from "react";
import { fetchAllProducts } from "../../../services/products";
import { HomePageProps, Product } from "@/types";
import AllProductsPage from "./AllProductsPage";

const AllProductsPageProvider: FC<HomePageProps> = ({
  initialProducts,
  totalPages,
}) => {
  const [currentPage, setCurrentPage] = useState(1);
  const [products, setProducts] = useState<Product[]>(initialProducts);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  const fetchData = async (page: number) => {
    setLoading(true);
    try {
      const allProducts = await fetchAllProducts(page);
      if (allProducts) {
        setProducts(allProducts.results);
        setCurrentPage(page);
      }
    } catch (error) {
      console.log("Error fetching products: ", error);
      setError(true);
    }
    window.scrollTo({ top: 0, behavior: "smooth" });
    setLoading(false);
  };

  return (
    <AllProductsPage
      loading={loading}
      error={error}
      products={products}
      currentPage={currentPage}
      totalPages={totalPages}
      fetchData={fetchData}
    />
  );
};

export default AllProductsPageProvider;
