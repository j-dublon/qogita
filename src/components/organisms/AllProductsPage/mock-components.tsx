import { AllProductsPageProps } from "@/types";
import React, { FC } from "react";

export const MockLoading: FC = () => <div>Loading...</div>;

export const MockError: FC = () => <div>Error</div>;

export const MockProductsList: FC = () => <div>Products List</div>;

export const MockPagination: FC = () => <div>Pagination</div>;

export const MockAllProductsPage: FC<AllProductsPageProps> = ({
  fetchData,
}) => (
  <div>
    Products Page
    <div onClick={() => fetchData(1)}>Products Page Pagination</div>
  </div>
);
