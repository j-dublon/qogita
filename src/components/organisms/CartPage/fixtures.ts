import { CartPageProps } from "@/types";

export const mockProps: CartPageProps = {
  items: [
    {
      name: "Mock Product",
      imageUrl: "http://your-product.com",
      recommendedRetailPrice: 52,
      price: 52,
      recommendedRetailPriceCurrency: "EUR",
      gtin: "987",
      id: "987",
      brandName: "Chanel",
      categoryName: "Perfume",
      quantity: 3,
    },
    {
      name: "Mock Product Three",
      imageUrl: "http://your-product-three.com",
      recommendedRetailPrice: 12,
      price: 12,
      recommendedRetailPriceCurrency: "EUR",
      gtin: "123",
      id: "123",
      brandName: "Lynx",
      categoryName: "Deodorant",
      quantity: 2,
    },
  ],
  handleUpdateCart: jest.fn(),
  cartTotal: 356,
};
