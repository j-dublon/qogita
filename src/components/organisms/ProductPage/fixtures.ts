import { Product, ProductPageProps } from "@/types";
import { Item } from "react-use-cart";

export const mockAddItem = jest.fn();
export const mockUpdateItemQuantity = jest.fn();

export const mockProps: ProductPageProps = {
  product: {
    name: "Mock Product",
    imageUrl: "http://your-product.com",
    recommendedRetailPrice: 52,
    recommendedRetailPriceCurrency: "GBP",
    gtin: "987",
    brandName: "Chanel",
    categoryName: "Perfume",
  },
  numberInCart: 3,
  updateCartQuantity: jest.fn(),
};

export const mockCartItem = {
  ...mockProps.product,
  id: mockProps.product.gtin,
  price: mockProps.product.recommendedRetailPrice,
  quantity: 3,
};

export const alternativeProduct: Product = {
  name: "Other Mock Product",
  imageUrl: "http://your-product-two.com",
  recommendedRetailPrice: 35,
  recommendedRetailPriceCurrency: "GBP",
  gtin: "073",
  brandName: "Chanel",
  categoryName: "Perfume",
};

export const alternativeCartItem: Item = {
  name: "Other Mock Product",
  imageUrl: "http://your-product-two.com",
  recommendedRetailPrice: 35,
  price: 35,
  recommendedRetailPriceCurrency: "GBP",
  gtin: "073",
  id: "073",
  brandName: "Chanel",
  categoryName: "Perfume",
};
