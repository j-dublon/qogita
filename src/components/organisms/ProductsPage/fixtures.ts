import { Product, ProductsPageProps } from "@/types";

const mockProducts: Product[] = [
  {
    name: "Parodontax Duplo Herbal Fresh 75ml",
    gtin: "5054563079435",
    recommendedRetailPrice: 29.99,
    recommendedRetailPriceCurrency: "EUR",
    imageUrl:
      "https://images.qogita.com/files/images/variants/aB9r5isuPDUTTD3nLNsXvQ.jpg",
    brandName: "Parodontax",
    categoryName: "Toothpaste",
  },
  {
    name: "Poseidon The Black Men Edt Vapo 150 Ml - Beauty & Health",
    gtin: "8411047151242",
    recommendedRetailPrice: 22.99,
    recommendedRetailPriceCurrency: "EUR",
    imageUrl:
      "https://images.qogita.com/files/images/variants/co8e7Y9gf272e2W2LgA6fj.jpg",
    brandName: "Instituto Espanol",
    categoryName: "Men's Perfume",
  },
  {
    name: "Snowracer Classic - Sled - Black",
    gtin: "7313327300382",
    recommendedRetailPrice: 90.49,
    recommendedRetailPriceCurrency: "EUR",
    imageUrl:
      "https://images.qogita.com/files/images/variants/jLSEtd5DRU72VXf7ScZ2m9.jpg",
    brandName: "Stiga",
    categoryName: "Sledding",
  },
];

export const mockProps: ProductsPageProps = {
  loading: false,
  error: false,
  products: mockProducts,
  currentPage: 1,
  totalPages: 5,
  fetchData: jest.fn(),
};

export const mockPropsLoading: ProductsPageProps = {
  ...mockProps,
  loading: true,
};

export const mockPropsErrorState: ProductsPageProps = {
  ...mockProps,
  error: true,
};
