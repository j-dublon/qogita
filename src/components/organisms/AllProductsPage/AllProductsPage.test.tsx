import React from "react";
import { render } from "@testing-library/react";
import AllProductsPage from "./AllProductsPage";
import { mockProps, mockPropsErrorState, mockPropsLoading } from "./fixtures";
import {
  MockError,
  MockLoading,
  MockPagination,
  MockProductsList,
} from "./mock-components";

jest.mock("../../atoms/Error/Error", () => ({
  __esModule: true,
  default: () => <MockError />,
}));
jest.mock("../../atoms/Loading/Loading", () => ({
  __esModule: true,
  default: () => <MockLoading />,
}));
jest.mock("../../molecules/ProductsList/ProductsList", () => ({
  __esModule: true,
  default: () => <MockProductsList />,
}));
jest.mock("react-responsive-pagination", () => ({
  __esModule: true,
  default: jest.fn(() => <MockPagination />),
}));
jest.mock("react-responsive-pagination/themes/classic.css", () => {});

describe("Component: AllProductsPage", () => {
  it("SHOULD match snapshot", () => {
    const component = render(<AllProductsPage {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it("SHOULD render title and components as expected WHEN page is not loading or in error state", () => {
    const { getByText, getAllByText } = render(
      <AllProductsPage {...mockProps} />
    );
    expect(getAllByText("Products")).toHaveLength(2);
    expect(getByText("Products List"));
    expect(getByText("Pagination"));
  });

  it("SHOULD render loading component WHEN loading is true", () => {
    const { getByText } = render(<AllProductsPage {...mockPropsLoading} />);
    expect(getByText("Loading..."));
  });

  it("SHOULD render error component WHEN error is true", () => {
    const { getByText } = render(<AllProductsPage {...mockPropsErrorState} />);
    expect(getByText("Error"));
  });
});
