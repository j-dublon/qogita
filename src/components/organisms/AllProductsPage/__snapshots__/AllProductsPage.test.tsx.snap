// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AllProductsPage SHOULD match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto px-4"
      >
        <div
          class="flex justify-between"
        >
          <strong>
            Qogita
          </strong>
          <nav>
            <ul
              class="flex gap-4"
            >
              <li>
                <a
                  class="underline"
                  href="/"
                >
                  Products
                </a>
              </li>
              <li>
                <a
                  class="underline"
                  href="/cart"
                >
                  Your Cart
                </a>
              </li>
            </ul>
          </nav>
        </div>
        <h1
          class="text-3xl font-semibold text-center my-6"
        >
          Products
        </h1>
        <div
          class="mb-8"
        >
          <div>
            Products List
          </div>
          <div>
            Pagination
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto px-4"
    >
      <div
        class="flex justify-between"
      >
        <strong>
          Qogita
        </strong>
        <nav>
          <ul
            class="flex gap-4"
          >
            <li>
              <a
                class="underline"
                href="/"
              >
                Products
              </a>
            </li>
            <li>
              <a
                class="underline"
                href="/cart"
              >
                Your Cart
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <h1
        class="text-3xl font-semibold text-center my-6"
      >
        Products
      </h1>
      <div
        class="mb-8"
      >
        <div>
          Products List
        </div>
        <div>
          Pagination
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
